src/HttpClientRequestAdapter.cs

```diff
@@ -216,12 +237,28 @@ public async Task SendNoContentAsync(RequestInformation requestInfo, IResponseHa
             requestInfo.Content?.Dispose();
             if(responseHandler == null)
             {
-                await ThrowFailedResponse(response, errorMapping);
-                response.Dispose();
+                try {
+                    await ThrowFailedResponse(response, errorMapping);
+                } finally {
+                    await purgeAsync(response);
+                }
             }
             else
                 await responseHandler.HandleResponseAsync<HttpResponseMessage, object>(response, errorMapping);
         }
+        private async Task purgeAsync(HttpResponseMessage response)
```

Comment:
C# terminology is usually "drain" for this scenario. Well actually, there are two scenarios, draining+dispose the response content and disposing entire response.

```suggestion
        private async Task drainAsync(HttpResponseMessage response)
```

I am [curious about this content](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.responsedraintimeout?view=net-6.0):

> Draining occurs when a request is cancelled or a **response is disposed prior to fully reading the content**. If the time exceeds the value, the connection will be closed rather than reused.

Safer to do it ourselves to make sure.

Reply:
yep, and it also depends on which handler is in use, which depends on the underlaying runtime.
renamed the method.